### PR TITLE
Ensure consistent memory stats snapshot by a process.

### PR DESCRIPTION
**Pull request by geirst**
@toregge please review
@baldersheim FYI

### CODE REVIEW

File: staging_vespalib/src/vespa/vespalib/util/process_memory_stats.cpp [hunk 1]
```diff
@@ -143,4 +124,76 @@ ProcessMemoryStats::create()
...
+      _mappings_count(mappings_cnt)
+{
+}
+
+namespace {
+
+bool
+similar(uint64_t lhs, uint64_t rhs, uint64_t epsilon)
+{
+    return (lhs < rhs) ? ((rhs - lhs) < epsilon) : ((lhs - rhs) < epsilon);
```
Comment (toregge) [inline staging_vespalib/src/vespa/vespalib/util/process_memory_stats.cpp]: Please use <=, so we get expected behavior if epsilon is 0.

File: staging_vespalib/src/tests/util/process_memory_stats/process_memory_stats_test.cpp [hunk 2]
```diff
@@ -27,7 +27,7 @@ std::string toString(const ProcessMemoryStats &stats)
 
 TEST("Simple stats")
 {
-    ProcessMemoryStats stats(ProcessMemoryStats::create());
+    ProcessMemoryStats stats(ProcessMemoryStats::create(4096));
```
Comment (toregge) [inline staging_vespalib/src/tests/util/process_memory_stats/process_memory_stats_test.cpp]: The magic number 4096 is used multiple times, please use a named constant expression.

**Review by toregge — APPROVED** [review]
LGTM

**Comment by geirst** [comment]
@toregge PTAL

**Review by toregge — APPROVED** [review]
LGTM

**Pull request merged**